winch/codegen/src/codegen/mod.rs

```diff
@@ -77,9 +77,82 @@ where
     fn emit_start(&mut self) -> Result<()> {
         self.masm.prologue();
         self.masm.reserve_stack(self.context.frame.locals_size);
+
+        // Once we have emitted the epilogue and reserved stack space for the locals, we push the
+        // base control flow block.
+        self.control_frames
+            .push(ControlStackFrame::function_body_block(
+                self.sig.result,
+                self.masm,
+                &mut self.context,
+            ));
         Ok(())
     }
 
+    /// Handles else or end instructions when the compiler has entered into an unreachable
+    /// code state. These instructions must be observed to determine if the reachability
+    /// state should be restored.
+    ///
+    /// When the compiler is in an unreachable state, all the other instructions are not
+    /// visited.
+    pub fn handle_unreachable_else_or_end_operator(&mut self, op: &Operator) {
```

Comment:
If this is only ever used for `End` and `Else` and never anything else, then it may be worth splitting this into two functions?

Reply:
Agreed, let's split these cases out, like the other helpers.